wavenet/audio_reader.py

```diff
@@ -105,14 +169,22 @@ def thread_main(self, sess):
                 if self.sample_size:
                     # Cut samples into fixed size pieces
                     buffer_ = np.append(buffer_, audio)
-                    while len(buffer_) > self.sample_size:
+                    while len(buffer_) > 0:
```

Comment:
Hi @fehiepsi  No, I don't think there's any need for padding. The WaveNet can operate on any length sample. The sample_size just put an upper limit on how big any one sample would be so that memory required could be limited to match available memory.

A problem with the previous implementation is we would never enqueue the sample until it was at least sample_size long. Almost all files are shorter than 100000 samples. So the logic would keep appending wave files to the buffer until it got up to 100000 samples, and then enqueue. And all that after silence trimming. This creates a discontinuity when going from one file's waveform to the next. So we were training it to learn a discontinuity. I think that might be why many of the samples we've been generating have these pops. We've been training the model to create them.

The other reason for the change is that with global conditioning, we must specify a single speaker id when we give the model a waveform to train on. We can't give it a waveform that is an amalgamation of several different files, since each one in general has a different speaker id. Which global id would we use in that case?